### PR TITLE
Add https sync support (fix #149)

### LINKED ISSUE #149: HTTPS sync
The Grabbit sample config files specifies `serverHost` and `serverPort`. Our AEM instances are running https ports, so I would like to know if it is possible to tell Grabbit to use https instead of http.

### PR DESCRIPTION
**Pull request by jonekdahl**
Introduces support for https via a `serverScheme` configuration parameter, allowed values are `http` or `https`. The parameter defaults to `http` so it should be backwards compatible.

Nothing has been changed in grabbit.sh as I wasn't sure how to best implement a flag for disabling certificate checking.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/TWCable/grabbit/pull/154" target="_blank">ReviewNinja

**Comment by jbornemann** [comment]
Great work @jonekdahl! Thank you!!

We may be able to add https://curl.haxx.se/docs/manpage.html#-k to the grabbit.sh curl commands.

File: src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy [hunk 1]
```diff
@@ -151,6 +155,25 @@ class GrabbitConfiguration {
     }
 
 
+    private static String schemeVal(Map<String, String> configMap, String key,
```
Comment (jbornemann) [inline src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy]: I wonder if we should add some coverage around this in GrabbitConfigurationSpec?

File: src/test/groovy/com/twcable/grabbit/client/batch/steps/http/CreateHttpConnectionTaskletSpec.groovy [hunk 2]
```diff
@@ -157,4 +158,17 @@ class CreateHttpConnectionTaskletSpec extends Specification {
         then:
         uri.toString() == "http://localhost:4503/grabbit/content?path=%252Fcontent%252Ftest&after=2008-09-22T13%253A57%253A31.2311892-04%253A00&excludePath=%252Fexclude&excludePath=%252Fexclude%252Fmetoo"
     }
+
+    def "buildURIForRequest() builds a Grabbit https URI correctly"() {
```
Comment (jbornemann) [inline src/test/groovy/com/twcable/grabbit/client/batch/steps/http/CreateHttpConnectionTaskletSpec.groovy]: 👍

File: src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy (hunk 1 above)
Comment (jonekdahl) [inline src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy]: Good suggestion, I added a few tests as you suggested. Only in YAML format though, do you think that is sufficient?

File: src/test/groovy/com/twcable/grabbit/client/GrabbitConfigurationSpec.groovy [hunk 3]
```diff
@@ -163,6 +163,89 @@ class GrabbitConfigurationSpec extends Specification {
 
     }
 
+    def "Should return http server scheme by default"() {
```
Comment (jbornemann) [inline src/test/groovy/com/twcable/grabbit/client/GrabbitConfigurationSpec.groovy]: 👍

File: src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy (hunk 1 above)
Comment (jbornemann) [inline src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy]: Yeah, that should cover the function sufficiently.

**Comment by jbornemann** [comment]
Pulling to try out the changes..

**Comment by jbornemann** [comment]
Looks good for http case! I don't have certificate stores, keys, etc set up properly for SSL. Did everything look fine when you tested in your setup? 

A last note, is maybe we should add a line to 

https://github.com/TWCable/grabbit/blob/master/docs/Running.adoc

describing serverScheme as an optional field. 

Thanks! We will get this merged!

**Comment by jonekdahl** [comment]
Hi @jbornemann. Yes, I have verified the https case. I also pushed a minor documentation patch adding a description for `serverScheme`, I hope that covers it?

**Comment by jbornemann** [comment]
That should cover it @jonekdahl! Thank you for your contribution! 😄 

I'll get this merged.

File: src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy (hunk 1 above)
Comment (jbornemann) [inline src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy]: !resolved

File: src/test/groovy/com/twcable/grabbit/client/GrabbitConfigurationSpec.groovy (hunk 3 above)
Comment (jbornemann) [inline src/test/groovy/com/twcable/grabbit/client/GrabbitConfigurationSpec.groovy]: !resolved

File: src/test/groovy/com/twcable/grabbit/client/batch/steps/http/CreateHttpConnectionTaskletSpec.groovy (hunk 2 above)
Comment (jbornemann) [inline src/test/groovy/com/twcable/grabbit/client/batch/steps/http/CreateHttpConnectionTaskletSpec.groovy]: !resolved

**Comment by sagarsane** [comment]
Awesome 👍 thanks for the contribution @jonekdahl!

**Comment by jbornemann** [comment]
Squashed and merged! Thanks again!

**Pull request closed without merging**